src/data/finset/lattice.lean

```diff
@@ -410,16 +409,20 @@ lemma sup_inf_distrib_right (s : finset ι) (f : ι → α) (a : α) :
   s.sup f ⊓ a = s.sup (λ i, f i ⊓ a) :=
 by { rw [_root_.inf_comm, s.sup_inf_distrib_left], simp_rw _root_.inf_comm }
 
-protected lemma disjoint_sup_right : disjoint a (s.sup f) ↔ ∀ i ∈ s, disjoint a (f i) :=
-by simp only [disjoint_iff, sup_inf_distrib_left, sup_eq_bot_iff]
+protected lemma disjoint_sup_right : disjoint a (s.sup f) ↔ ∀ ⦃i⦄, i ∈ s → disjoint a (f i) :=
```

Comment:
These changes are worth a remark in the description

Reply:
Please also explain *why* you did this!

Reply:
Isn't it clear? You don't want to end up with overly explicit binders after rewriting a local assumption.

Reply:
I think that argument probably leads you replacing every `∀ i ∈ s,` in mathlib with `∀ ⦃i⦄, i ∈ s →`...

Reply:
Not quite all, no. But, yes, I think that's a good thing.

Reply:
Was there ever a Zulip thread about this?

Reply:
Probably not about iffs specifically. But it really is just an extension of the rule we've been applying elsewhere. Note that the reason I'm modifying this one in particular is because the one side implication isn't stated as a separate lemma with `i` an implicit argument. And since changing the binder implicitness fixes my use case without introducing a redundant lemma, we might as well.